Public/Src/Tools/VBCSCompilerLogger/VBCSCompilerLogger.cs

```diff
@@ -0,0 +1,191 @@
+        }
+
+        private void EventSourceOnMessageRaised(object sender, BuildMessageEventArgs e)
+        {
+            if (e is TaskCommandLineEventArgs commandLine)
+            {
+                // We are only interested in CSharp and VisualBasic tasks
+                string language;
+                string arguments;
+                switch (commandLine.TaskName)
+                {
+                    case CscTaskName:
+                        language = LanguageNames.CSharp;
+                        arguments = GetArgumentsFromCommandLine(CscToolName, commandLine.CommandLine);
+                        break;
+                    case VbcTaskName:
+                        language = LanguageNames.VisualBasic;
+                        arguments = GetArgumentsFromCommandLine(VbcToolName, commandLine.CommandLine);
+                        break;
+                    default:
+                        return;
+                }
+
+                // We were able to split the compiler invocation from its arguments. This is the indicator
+                // that something didn't go as expected. Since failing to parse the command line means we
+                // are not catching all inputs/outputs properly, we have no option but to fail the corresponding pip
+                if (arguments == null)
+                {
+                    throw new ArgumentException($"Unexpected tool name in command line. Expected '{CscToolName}' or '{VbcToolName}', but got: {commandLine.CommandLine}");
+                }
+
+                var parsedCommandLine = CompilerUtilities.GetParsedCommandLineArguments(language, arguments, commandLine.ProjectFile);
+                RegisterAccesses(parsedCommandLine);
+            }
+        }
+
+        private string GetArgumentsFromCommandLine(string toolToTrim, string commandLine)
+        {
+            toolToTrim += " ";
+            int index = commandLine.IndexOf(toolToTrim, StringComparison.OrdinalIgnoreCase);
+            
+            if (index == -1)
+            {
+                return null;
+            }
+
+            return commandLine.Substring(index + toolToTrim.Length);
+        }
+
+        private void RegisterAccesses(CommandLineArguments args)
+        {
+            // All inputs
+            RegisterInputs(args.SourceFiles.Select(source => source.Path));
+            RegisterInputs(args.AnalyzerReferences.Select(reference => reference.FilePath));
+            RegisterInputs(args.EmbeddedFiles.Select(embedded => embedded.Path));
+            RegisterInput(args.Win32ResourceFile);
+            RegisterInput(args.Win32Icon);
+            RegisterInput(args.Win32Manifest);
+            RegisterInputs(args.AdditionalFiles.Select(additional => additional.Path));
+            RegisterInputs(args.MetadataReferences.Select(metadata => metadata.Reference).Where(pathOrAssemblyName => FileUtilities.FileSystem.IsPathRooted(pathOrAssemblyName)));

```

Comment:
Why only add the rooted paths?

Reply:
Given the spec that string can be either an absolute path or an assembly display name. Looking at that more carefully, I don't really know how to handle the assembly display name since it needs some resolution that doesn't look like a static thing. Looking...

---
In reply to: [346068155](https://github.com/microsoft/BuildXL/pull/1207#discussion_r346068155) [](ancestors = 346068155)

Reply:
Ended up implementing a logic similar to the one the compiler does. SearchPaths are a collection of directories that are used for resolving relative paths

---
In reply to: [346076348](https://github.com/microsoft/BuildXL/pull/1207#discussion_r346076348) [](ancestors = 346076348,346068155)